File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        ans=[]
        for i in range(len(l)):
            v = nums[l[i]:r[i]+1]
            v.sort()
            temp = True
            if len(v) > 1:
                dif = v[1] - v[0]
                for j in range(1,len(v)) : 
                    if v[j] - v[j-1] != dif:
                        temp = False
                        break
            else:
                temp = False
            ans.append(temp)
        return ans
        

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        ArrayList<Boolean> ans = new ArrayList <>();
        for (int i = 0; i < l.length; i++){
            int[] v = Arrays.copyOfRange(nums, l[i], r[i]+1);
            Arrays.sort(v);
            boolean temp = true;
            if (v.length > 1){
                int dif = v[1] - v[0];
                for (int j = 1; j < v.length; j++){
                    if(v[j] - v[j-1] != dif){
                        temp = false;
                        break;
                    }
                }
            }
            else{
                temp = false;
            }
            ans.add(temp);
        }
        return ans;
    }
}